src/drivers/imx/interrupt.c

```diff
@@ -180,12 +218,28 @@ const char * const irq_name_irqsteer[] = {
 	"irqsteer6",
 	"irqsteer7"
 };
+#endif
 
 #define IRQ_MAX_TRIES	1000
 
 /* Extract the 64 status bits corresponding to output interrupt line
  * index (64 input interrupts)
  */
+#ifdef CONFIG_IMX8M
+static uint64_t get_irqsteer_interrupts(uint32_t index)
+{
+	uint64_t result = 0;
+
+	result = irqstr_get_status_word(2 * index);
+	/* line 0 is special only maps interrupts [0..31] */
+	if (index == 0)
+		return result;
```

Comment:
For correctness isn't it necessary to move this return after the shift below, so that we can correctly shift interrupts 0-31 to 32-63 like the others and allow for the correction to shift back?

Reply:
Yes, you are right will fix.